runtime/lua/vim/treesitter.lua

```diff
@@ -658,4 +658,36 @@ function M.foldexpr(lnum)
   return require('vim.treesitter._fold').foldexpr(lnum)
 end
 
+---Get the lang for a buffer and position taking into account
+---language injections.
+---@param opts table|nil Optional keyword arguments:
+---             - bufnr integer|nil Buffer number (nil or 0 for current buffer)
+---             - pos table|nil 0-indexed (row, col) tuple. Defaults to cursor position in the
+---                             current window. Required if {bufnr} is not the current buffer
+---@return string|nil
+function M.get_lang(opts)
```

Comment:
This function already exists elsewhere:
https://github.com/neovim/neovim/blob/d1e0f7454b5fc61d26db5af5ce00c1894e7c49fc/runtime/lua/vim/treesitter/language.lua#L8-L12

(Guess who added that last month ;))

Reply:
(Can probably be combined into one interface by adding `filetype` to the `opts` keys here and marking the other function as private via `_get_lang_from_ft`.)

Reply:
`get_lang()` gets removed in #22643. Instead, we will use `vim.filetype.get_option(filetype, 'parser')` as a direct replacement.

Reply:
FYI, this is why I was trying to drill down if `ftplugin` was the way to go to use as a namespace.

Reply:
> get_lang() gets removed in https://github.com/neovim/neovim/pull/22643. Instead, we will use vim.filetype.get_option(filetype, 'parser') as a direct replacement.

So that PR is a prerequisite for this one? (Asking because that duplication is what's failing all the builds.)

Reply:
Also, I'm planning on changing the nomenclature to be "parser" instead of "lang". TS uses "language" but I think that's too confusing with out "filetype". I think it's much easier to understand the concept of a "parser" being able to parser multiple "filetypes".

Reply:
As long as people coming from treesitter can make the connection (and vice versa, if someone needs to look up upstream docs).

Reply:
Good point actually, because TS already has the concept of "parser". God dammit.

Are we ok adding a `'lang'` bufoption in the presence of `'syntax'` and `'filetype'`?

Reply:
Not ideal; I did prefer `parser` for it. I wish we had native table options already...

(Also, remember that it's n by m: multiple parsers may parse a filetype, and you might use different ones for different buffers -- for example upstream C and our forked C.)

But I don't really have a better suggestion at the moment.

Reply:
Ok, I'll stick with `parser` for now. It's ok if the user mentally thinks of this as a TS parser.